### PR TITLE
Route53 client

### PR DESCRIPTION
**Pull request by misantron**
#997

### CODE REVIEW

**Comment by Nyholm** [comment]
Cool, well done. Remember to run `composer dump-autoload`, that will make sure your local installation is aware of the new files. 
Now you may generate the "operations" (client methods) you need. Start with just one or two and then we can follow up with more (if you need it) in the next PR. 

There is also a lot of tests to make sure you update the SymfonyBundle, Service factory, Readme, Docs etc.

**Comment by GromNaN** [comment]
👍🏻 You can add it to the readme too.
https://github.com/async-aws/aws#packages-overview

File: src/Service/Route53/Makefile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.EXPORT_ALL_VARIABLES:
+
+initialize: start-docker
+start-docker:
+	echo "Noop"
```
Comment (jderusse) [inline src/Service/Route53/Makefile:5]: LocalStack provides a service for Route53

You can take inspiration from `SSM` (see `Makefile`, `src/Service/Ssm/Makefile` and `SsmClientTest::getClient`)

Comment (misantron) [reply src/Service/Route53/Makefile:5]: Nice, I'll take a look

**Comment by misantron** [comment]
@Nyholm @jderusse Can anybody approve CI run please? Want to see progress in tests and other checks

**Comment by Nyholm** [comment]
I see that there are a few tests still failing.

**Comment by Nyholm** [comment]
Hm.. It looks like we get 500 errors. is that due to the docker image?

File: docs/integration/symfony-bundle.md [hunk 2]
```diff
@@ -44,6 +44,7 @@ The config above will create the following services:
 | `async_aws.client.sqs`     | AsyncAws\Sqs\SqsClient
 | `async_aws.client.delayed` | AsyncAws\Sqs\SqsClient $delayed
 | `async_aws.client.s3`      | AsyncAws\S3\S3Client
+| `async_aws.client.route53` | AsyncAws\Route53\Route53Client
```
Comment (jderusse) [inline docs/integration/symfony-bundle.md:47]: to remove

File: Makefile [hunk 3]
```diff
@@ -20,7 +20,7 @@ start-docker: start-docker-s3 start-docker-localstack
 start-docker-localstack:
 	docker start async_aws_localstack && exit 0 || \
 	docker pull localstack/localstack && \
-	docker run -d -p 4566:4566 -p 4567:4566 -p 4568:4566 -p 4571:4566 -p 4572:4566 -p 4573:4566 -p 4574:4566 -p 4575:4566 -e SERVICES=sts,cloudformation,logs,events,iam,sns,ssm,dynamodb -v /var/run/docker.sock:/var/run/docker.sock --name async_aws_localstack localstack/localstack && \
+	docker run -d -p 4566:4566 -p 4567:4566 -p 4568:4566 -p 4571:4566 -p 4572:4566 -p 4573:4566 -p 4574:4566 -p 4575:4566 -p 4580:4566 -e SERVICES=sts,cloudformation,logs,events,iam,sns,ssm,dynamodb,route53 -v /var/run/docker.sock:/var/run/docker.sock --name async_aws_localstack localstack/localstack && \
```
Comment (jderusse) [inline Makefile:23]: please use `4576`

**Comment by jderusse** [comment]
I created #1004 to help you to bootstrap the client.

**Comment by jderusse** [comment]
Could you rebase on top of master? The code-review will be easiest

File: src/Service/Route53/tests/Unit/Input/ChangeResourceRecordSetsRequestTest.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            ]),
+        ]);
+
+        // see example-1.json from SDK
+        $expected = '
+POST /2013-04-01/hostedzone/Z1D633PEXAMPLE/rrset/ HTTP/1.0
+Content-Type: application/xml
+';
+
+        self::assertRequestEqualsHttpRequest($expected, $input->request());
```
Comment (jderusse) [inline src/Service/Route53/tests/Unit/Input/ChangeResourceRecordSetsRequestTest.php:52]: Is it true? I would expect X:L content in the body... That's weird the test is not failling

Comment (jderusse) [reply src/Service/Route53/tests/Unit/Input/ChangeResourceRecordSetsRequestTest.php:52]: oh! it looks like the generator is not able to generate code for RestXML  (until now, we only deal with S3 and file's content in BODY

https://github.com/async-aws/aws/blob/e2edb3dd266aa76ef2ab87040209728fe9c655ff/src/CodeGenerator/src/Generator/RequestSerializer/RestXmlSerializer.php#L37-L39

I'll open a PR to fix that.

**Comment by Nyholm** [comment]
This PR was way more tricky than I thought. 

Now when @jderusse added some work. If we rebase this will everything work then?

**Comment by jderusse** [comment]
I rebased your PR and push commit to fix falling tests. Please `pull` my changes before working on it.

@Nyholm this PR is now RFR

**Review by Nyholm — APPROVED** [review]
Im happy with this

**Pull request merged**